core/trino-main/src/main/java/io/trino/operator/join/unspilled/JoinProbe.java

```diff
@@ -30,39 +31,60 @@
 import static io.trino.spi.type.BigintType.BIGINT;
 import static java.util.Objects.requireNonNull;
 
+/**
+ * This class eagerly calculates all join positions and stores them in an array
+ * OR
+ * stores a single RLE position in a single value array
+ * The fetching of the data is uniform using by using & operator on the current index.
+ * This is more performant that checking whether values are rle/non-rle every time.
+ */
 public class JoinProbe
```

Comment:
how does it prevent `PageJoiner` from visiting all probe rows?

Reply:
It doesn't. Classes are isolated so this is getting inlined. Static function invocations were never a bottleneck here.

It doesn't mean that rewriting the whole PageJoiner won't bring other benefits.